### ISSUE TITLE
No option to delete the history

### ISSUE THREAD

**Issue by 29ayush** [cfeenstra67/egghead#1]
Thanks for the add-on. The search tags seems amazing. 
The session seems not useful thou. In any case, I would like to request two features:
- Delete an individual item from history
- Delete all entries for a searched item

**Comment by cfeenstra67** [comment]
Hey there @29ayush , thanks for opening an issue. I can definitely look into this when I have some time over the next couple of weeks; if you'd like to submit a PR before then please feel free, I'll review it.

**Comment by cfeenstra67** [comment]
Hey there @29ayush , took me a while but I got the chance to revisit this project. Yesterday I released version 2.0.0 that does include history deletion features like the ones I think you were looking for. If you have Egghead installed already it should be updated automatically in the next couple days, or you can remove + reinstall the extension to get the latest version right away. If you have a chance to try it out and have any feedback, please let me know!

**Comment by 29ayush** [comment]
Hey Cam, 
Thanks for the udpate.
Haha, I think I went overboard when typing this feedback, but well, since I already typed, apologies :) 

TL;DR:- Better placement of delete buttons, and some documentation.

Overall feedback for the extension:

Currently the UI seems a little less intuitive and accessible, I couldn't even find the Actions->Delete until I fiddled with everything

- Delete, and Details button should both be more visible.
- Same goes for reset filters, and Delete Sessions 

I have tried another extension `BetterHistory` and it's UI seems good in some aspects. Maybe you can also look at it for inspiration.I don't want to blatantly copy suggestions from it, but it's placement of the delete buttons is really good.

----

- Navigation is quite problematic. 
- Especially if I go into details of a link, then I can't go back at all, probably a pop-up overlay instead of seperate url for details page, ( like you have for delete ). I would like to go back to same scroll and page after looking at the details.
- Highlight of Hosts can be made like that of keywords, and the other hosts should not disappear after clicking one. IF it's a technical limitation of search which doesn't allow multiple hosts, atlteast a white overlay instead of current subtle would be better imo, plus some indication that there are other hosts it's just that you have selected this one currently.  So like click to select a host, turns white, other remains disabled until you deselect white, or clicking other changes to the new one.
- I like the reset filters options, however as previously stated, it's very hidden currently.
Plus there should be a filter bar showing all the filters. Currently I see that you add a date bar if a date is selected. You can have other filters present in the same bar as well. so people know what filters are selected. ( I assume hosts, keyword, search, date are the current supported filters)
So 3 jan - 5 jan , www.google.com, anime, journey to   [Reset Filters ]
Something like this
If you don't want to display filters at two places it would still be a good idea, to atleast state that search, host, keyword filters are activated
So 3jan - 5Jan, Keywords (Highlighted -just title, no specifc keyword required.) , Hosts, Search

----
Probably a bug, or Idk what do you want it to show but I keep seeing a undefined next to timestamp in the history. 
I would prefer to see a full url. Again BetterHistory seems to have a better visibility on this, the time on the right, with full url below the title seem good.
If you personally are against full urls, maybe you can provide an option in settings for it.

For you, the right space is occupied by `x links` so I am not sure how much space you have there to put delete, details, and time ( maybe ) there, maybe you can follow that on hover the title display length gets shorter,( you don't need to wrap it just gets overlayed ), and , delete  button get added to the end instead of three dots, or along with it( future if you add something more ). Details can show up on left of links button.  An icon( for delete ) might even save more space.

<img width="654" alt="Image" src="https://github.com/user-attachments/assets/bdd3f116-8d67-4a8f-9978-fa80037bc8b4" />

P3: Random
- Maybe you can add a custom date range option. I personally don't use it much, but since you have already added concept of date ranges, maybe you can add this as well. In my suggested filter bar, a date column always shows, and it changes if you click on graph and clicking on filter area(date button ) allows to set custom date?

Couple of nits:
- I really don't comprehend the concept of sessions you have, maybe some documentation please. From just exploringI couldn't understand. And thus delete also seems a bit complicated. as it shows delete sessions. like what is a session. 
- I don't understand what is the `x links` represent. Opening it for me, just shows same website as sub children of the current item. I don't get it, please add documentation for it.

**Comment by cfeenstra67** [comment]
Hey there @29ayush 
I have a new version that's been pending review for a couple of days so should likely be approved soon (their review process seems to have gotten pretty slow--it's been taking ~a week or so to get new versions deployed).

I really do appreciate the feedback--it's been very difficult for me to get since this is a chrome extension + slightly niche.

I'll let you know when it's published, but I think this new version should address most of your points of feedback:
- Delete + Detail Buttons removed from dropdowns and made more obvious
- The "reset" and "delete" controls at the top of search were also removed from the dropdown so they are more obvious
- Opening detalis from search will now be a modal instead of navigating pages
- Added a setting for showing the full URL in search rather than just the host
- Addressed all of your points of feedback on the filters I think--now all applied filters will appear at the top of the search and can be removed one-by-one or reset w/ the "reset" button. I also made it so that the "host" options will not be filtered by the currently selected host so you can select multiple; I didn't fully understand your first sentence though: "Highlight of Hosts can be made like that of keywords"--what did you mean by that?
- Fixed the issue where it shows undefined next to the timestamp--I think you're using 24 hour/"military" time, right?
- Added "Search help" (tool inspiration from Better History on this!) w/ some basic info about the things you can do w/ searches. I didn't add an explicit date filter in the end, but you can achieve that through the search syntax for when it is needed.

A number of other small improvements as well.

To address your question about "sessions"--that's something that won't be fixed in this version, however I will probably try to fix it at some point. The important thing to understand is that the problem only appears from existing history Egghead "backfills" from your chrome history--as you use the extension going forward, a "session" should be pretty straightforward--a specific URL that you open in a specific tab at some point in time--the "links" should correspond to actual links you've opened from that session--unfortunately the current behavior w/ backfilling the history from results in lots of these weird "link" relationships between the same URL. That's partly a limitation of the information that Chrome stores, but it should ultimately be solvable in Egghead. That's a slightly bigger job though--but rest assured the behavior going forward should be correct. The reason that you're likely seeing this now is because Egghead 2.0 introduced some technical changes to how the data is stored, and so we have to re-backfill the data from chrome on the upgrade to 2.0. That should never need to happen again, though. Sorry for the bad experience there; I can add some info in the docs about this as well, but the confusing behavior you're seeing is mainly just a straightforward bug

In any case, I'll let you know when the latest version gets published, as I said it should be in the next couple of days hopefully--would really appreciate any follow-up comments you have!

**Comment by cfeenstra67** [comment]
Latest version got approved this morning, you should see the extension get updated in the next day or two; I'll close this issue in a few days if I don't hear from you, but if you have any more feedback I'm happy to hear it!

**Comment by 29ayush** [comment]
Thanks for so quickly addressing the feedback. I love this 2.2 much better than 2.0 definitely.

The filters, and details popup look pretty and I love it. I love the new Visited and Navigated tabs, were they there in 2.0. 
Not sure how I feel about you skipping the get parameters ( ?x=&y=) and only using the url for grouping in visited to  ( I think I like it, but not sure ) . 

I like the full url setting as well.

> The "reset" and "delete" controls at the top of search were also removed from the dropdown so they are more obvious

Yup they definitely are looking much better, feel still slightly off ( not sure, maybe it's just getting used to it ). Can't think of any suggestions for it. 

> "Highlight of Hosts can be made like that of keywords"--what did you mean by that?

I meant the highlighting of host in left pane is not very visible.
Light View:
<img width="259" alt="Image" src="https://github.com/user-attachments/assets/4e12e994-46dd-4f73-be50-2a0dac9f75aa" />

Dark view is slightly better but still too light, ( did you change the background for dark theme, I feel like you have chosen more black now, I prefered the former, maybe again needs some getting used to,  again I am not the fan of dark mode, I have only recently started using it, so not the best one for suggestion/advise here, I still prefer things a little light.  ) 
<img width="245" alt="Image" src="https://github.com/user-attachments/assets/e618df94-26a3-4201-ada6-f48180b62c1a" />

> Fixed the issue where it shows undefined next to the timestamp--I think you're using 24 hour/"military" time, right?

Yes thank you no longer seeing undefined, Regarding time format, (yes I am seeing 24 H format ) I am not even sure, how to change that. I don't see a option in your extension settings. 

-----

Nit picking:
Dark Mode;
<img width="1260" alt="Image" src="https://github.com/user-attachments/assets/d49b3ca0-a4a3-4cf4-9247-ff81cc187e79" />
The date view area should be slightly bolder ( different ), it currently is not emphasized enough imo.

Regarding timestamps:
I probably like this default from chrome a bit better, the timestamps within a day being seperately listed makes timeline more clearer. 

<img width="1035" alt="Image" src="https://github.com/user-attachments/assets/87d52513-4a4a-4dc3-98d2-d9f6796f215c" />

Personal Bias: I would move the dates instead of favicon, shift the favicons more right, ( like chromes default view ), same thing for url ( move it to side like default chrome, because when looking at history I don't know how important is the url, for each entry, especially if you can get it from details pop up  ).
Hmm.. 
Ok so, 
Default View: View it like chrome ( checkboxes are ok to have, not have, I dont' think they have much use, so hidden behind your global delete button is probably ok )
Settings Page ( Show Full Urls ) -> This moves urls from side to below, ( like you have now )  

-----

Feature Request:
- A day selector in the day section of the infinite scroll, choosing a date will bring that date to top. I think you can just have infinite scroll date ( no need to parse history to see if history is really prsent on a date), we can just move to a last date which exists before the selected date. 

Typing this I realized I asked for a date selector again, but I don't know, the new date viewing area makes me think there is an excellent oppurtunity for it.

<img width="1512" alt="Image" src="https://github.com/user-attachments/assets/448530b0-7b61-4489-bf0f-730919576784" />

**Comment by 29ayush** [comment]
> no need to parse history to see if history is really prsent on a date), we can just move to a last date which exists before the selected date.

I missed the case regarding search, where dates are further filtered, probably blind dates won't work.

**Comment by 29ayush** [comment]
1 more bug

<img width="94" alt="Image" src="https://github.com/user-attachments/assets/7c7e0836-55e8-41a2-97b9-dec2bb89b37d" />
<img width="978" alt="Image" src="https://github.com/user-attachments/assets/24ce3c4c-0363-4db8-ade5-5bef4f57037c" />

I see less than 3 letter words in #words section, but search doesn't work for <3 letter.

**Comment by cfeenstra67** [comment]
Thanks for the additional feedback @29ayush !! I haven't taken a close look yet but will get to it when I have some time
